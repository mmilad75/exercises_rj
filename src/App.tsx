import React, { useState } from "react";
import AppHeader from "./components/app-header/app-header.component";
import Counter from "./components/counter/counter.component";
import Login from "./components/login/login.component";
import { Components } from "./components/types";
import UsersList from "./components/users-list/users-list.component";
import styles from "./App.styles";

const App: React.FC = () => {
    const [component, setComponent] = useState<Components>("users");

    const renderComponent = () => {
        switch (component) {
            case "counter":
                return <Counter />;
            case "login":
                return <Login />;
            case "users":
                return <UsersList />;
        }
    };

    return (
        <div className="App">
            <AppHeader selected={component} onSelect={setComponent} />
            <div style={styles.componentsContainer}>{renderComponent()}</div>
        </div>
    );
};

export default App;
