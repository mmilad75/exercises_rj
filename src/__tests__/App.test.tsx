import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import React from "react";
import App from "../App";

describe("App", () => {
    it("should render correctly", () => {
        jest.spyOn(React, "useEffect").mockImplementationOnce(() => {});
        const { asFragment } = render(<App />);
        expect(asFragment()).toMatchSnapshot();
    });

    it("should navigate to counter", () => {
        render(<App />);
        const counterButton = screen.getByText("Counter");

        userEvent.click(counterButton);

        expect(screen.getByText("Counter: 0")).toBeInTheDocument();
    });

    it("should navigate to login", () => {
        jest.spyOn(React, "useEffect").mockImplementationOnce(() => {});

        render(<App />);
        const counterButton = screen.getByText("Login");

        userEvent.click(counterButton);

        expect(
            screen.getByPlaceholderText("your-email@email.com")
        ).toBeInTheDocument();
    });
});
