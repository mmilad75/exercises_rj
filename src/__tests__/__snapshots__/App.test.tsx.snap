// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <header
      role="navigation"
      style="display: flex; justify-content: center; background: rgb(226, 226, 226); margin-bottom: 20px; border-radius: 8px;"
    >
      <div
        role="button"
        style="margin-right: 16px; padding: 20px; cursor: pointer; color: rgb(29, 161, 242);"
      >
        Users
      </div>
      <div
        role="button"
        style="margin-right: 16px; padding: 20px; cursor: pointer;"
      >
        Counter
      </div>
      <div
        role="button"
        style="margin-right: 16px; padding: 20px; cursor: pointer;"
      >
        Login
      </div>
    </header>
    <div
      style="width: 90%; max-width: 960px; margin: 0px auto;"
    >
      <div>
        <h1>
          Users List
        </h1>
        <p>
          Loading users...
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
